sssd_test_framework/roles/ad.py

```diff
@@ -1490,4 +1545,216 @@ def __get_info(
         return info
 
 
+class GPO(BaseObject[ADHost, AD]):
+    """
+    Group policy object management.
+    """
+
+    def __init__(self, role: AD, name: str) -> None:
+        """
+        :param role: AD host object.
+        :type role:  ADHost
+        :param name: GPO name, defaults to 'Domain Test Policy'
+        :type name: str, optional
+        """
+        super().__init__(role)
+
+        self.name: str = name
+        """Group policy display name."""
+
+        self._search_base: str = f"cn=policies,cn=system,{self.role.host.naming_context}"
+        """Group policy search base."""
+
+        self._dn: str = self.get("DistinguishedName")
+        """Group policy dn."""
+
+        self._cn: str = self.get("CN")
+        """Group policy cn."""
+
+    def get(self, key: str) -> str:
+        """Get group policy attributes.
+
+        :param key: Attribute to get.
+        :type key: str
+        :return: Key value.
+        :rtype: str
+        """
+        result = self.role.host.ssh.run(
+            rf"""
+            $query = "(&(ObjectClass=groupPolicyContainer)(DisplayName={self.name}))"
+            Get-ADObject -SearchBase "{self._search_base}" -Properties "*" -LDAPFilter $query 
+            """
+        ).stdout_lines
+
+        i = 0
+        while i < len(result):
+            if result[i].startswith(key):
+                if result[i + 1].startswith(" "):
+                    value = (result[i].strip() + result[i + 1].strip()).split(":")
+                    return value[1].strip()
+                else:
+                    value = result[i].split(":")
+                    return value[1].strip()
+            i += 1
+
+    def delete(self) -> None:
+        """
+        Delete group policy object.
+
+        Note, GPLinks and GPO files are removed test during cleanup.
+        """
+        self.role.host.ssh.run(f'Remove-GPO -Guid "{self._cn}" -Confirm:$false')
+
+    def add(self) -> GPO:
+        """
+        Add a group policy object.
+
+        This creates the GPO and to update the security database, the SecEdit path is then
+        created with the GptTempl.inf file. The file populated with the generic headers but
```

Comment:
The file **is** populated?

Reply:
It is. 
`
$content = @{{'Unicode'=@{{'Unicode'='yes'}};'Version'=@{{'signature'='"$CHICAGO$"';'Revision'='1'}}}}
Out-IniFile -InputObject $content -FilePath $file
`

`
[Unicode]
Unicode=yes

[Version]
signature=$CHICAGO$
Revision=1
`

Relies on this https://github.com/SSSD/sssd-ci-containers/pull/83

Reply:
Oh, it was not a question, but suggestion to the text :-) There is "The file populated" (without "is")